### ISSUE TITLE
Add CVE IDs and CVSS scores to "GET /images/..../vuln" response records

### ISSUE THREAD

**Issue by nurmi** [anchore/anchore-engine#163]
Feature request is include additional information, when possible, in the response records of the image vuln scan result from v1/images/<digest>|by_id/<id>/vuln route.  Specific ask follows:

While some vulnerabilites happen to look like CVE IDs not all (e.g. RHSA) are. Anchore should have a flag to request that the response to "GET /images/by_id/{imageId}/vuln" included extra CVE data for each vulnerability. This should include the CVE ID, and CVSS V2 and CVSS V3 scores. If a particular CVE does not have score (possible when a CVE is still being evaluated or if it pre-dates the introduction of CVSS V3) then an "impossible" value should be returned (0, -1, "", etc).

**Comment by nurmi** [comment]
[We've been working on the addition of CVSS information in anchore-engine, alongside our upcoming update to ingress the JSON based NVD data feeds (much better/uniform CVSS information in that source data!).  We've got a functional prototype that is generating results, and I thought I would share some examples of how it will look here (and will update the issue #164 with policy examples as well):

output example of /images/..../vuln - note that the record is the same as existing API spec but with the addition of 'nvd_data' which will be populated iff the nvd data feed is enabled and synced.  If the NVD records are available but are missing scores (lots of partials in the NVD source), then the score value is set to -1.0.  The 'nvd_data' element is a list, to handle cases (like ELSA, RHSA, etc) where the top level vulnerability references multiple NVD CVEs.

```
{
    "imageDigest": "sha256:8b72525bf36671bead1b0aba00ffc784c2ed7ba95962ecb565830cb22dd1a087", 
    "vulnerabilities": [
        {
            "feed": "vulnerabilities", 
            "feed_group": "ol:7", 
            "vuln": "ELSA-2019-1880"
...
...
...
            "nvd_data": [
                {
                    "cvss_scores": {
                        "cvssV2": {
                            "baseScore": 10.0, 
                            "exploitabilityScore": 10.0, 
                            "impactScore": 10.0
                        }, 
                        "cvssV3": {
                            "baseScore": 9.8, 
                            "exploitabilityScore": 3.9, 
                            "impactScore": 5.9
                        }
                    }, 
                    "id": "CVE-2018-14618"
                }
            ], 
            ...
            ...
...
...
}
```

**Comment by nurmi** [comment]
Adding the (same 'nvd_data' structure) result example from the response to a GET to /query/vulnerabilities... route as well:

```
curl -u admin:foobar http://localhost:8228/v1/query/vulnerabilities?id=RHSA-2019:0679
{
  "page": "1",
  "returned_count": 1,
  "total_count": 1,
  "vulnerabilities": [
    {
      "affected_packages": [
        {
          "name": "libssh2",
          "type": "rpm",
          "version": "0:1.4.3-12.el7_6.2"
        },
        {
          "name": "libssh2-devel",
          "type": "rpm",
          "version": "0:1.4.3-12.el7_6.2"
        },
        {
          "name": "libssh2-docs",
          "type": "rpm",
          "version": "0:1.4.3-12.el7_6.2"
        }
      ],
      "id": "RHSA-2019:0679",
      "link": "https://access.redhat.com/errata/RHSA-2019:0679",
      "namespace": "centos:7",
      "nvd_data": [
        {
          "cvss_scores": {
            "cvssV2": {
              "baseScore": 9.3,
              "exploitabilityScore": 8.6,
              "impactScore": 10.0
            },
            "cvssV3": {
              "baseScore": 8.8,
              "exploitabilityScore": 2.8,
              "impactScore": 5.9
            }
          },
          "id": "CVE-2019-3855"
        },
        {
          "cvss_scores": {
            "cvssV2": {
              "baseScore": 6.8,
              "exploitabilityScore": 8.6,
              "impactScore": 6.4
            },
            "cvssV3": {
              "baseScore": 8.8,
              "exploitabilityScore": 2.8,
              "impactScore": 5.9
            }
          },
          "id": "CVE-2019-3856"
        },
        {
          "cvss_scores": {
            "cvssV2": {
              "baseScore": 6.8,
              "exploitabilityScore": 8.6,
              "impactScore": 6.4
            },
            "cvssV3": {
              "baseScore": 8.8,
              "exploitabilityScore": 2.8,
              "impactScore": 5.9
            }
          },
          "id": "CVE-2019-3857"
        },
        {
          "cvss_scores": {
            "cvssV2": {
              "baseScore": 6.8,
              "exploitabilityScore": 8.6,
              "impactScore": 6.4
            },
            "cvssV3": {
              "baseScore": 8.8,
              "exploitabilityScore": 2.8,
              "impactScore": 5.9
            }
          },
          "id": "CVE-2019-3863"
        }
      ],
      "severity": "High"
    }
  ]
}
```

**Comment by juledwar** [comment]
This looks excellent, thanks.

**Comment by zhill** [comment]
Addressed in the v0.5.0 release
